cpp/src/utilities/device_atomics.cuh

```diff
@@ -330,9 +230,9 @@ namespace detail {
         __forceinline__  __device__
         T operator()(T* addr, T const & update_value, DeviceMin op)
         {
-            using T_out = long long int;
-            T ret = atomicMin(reinterpret_cast<T_out*>(addr),
-                type_reinterpret<T_out, T>(update_value) );
+            using T_int = long long int;
+            T ret = atomicMin(reinterpret_cast<T_int*>(addr),
```

Comment:
So you are reinterpret casting a int64_t to a long long int. What if those are not the same thing? Why do you need this cast?

Reply:
I saw the discussion on the issue about why you need to use `long long int`. I think in the issue @jrhemstad  meant you should have a `static_assert(sizeof(T) == sizeof(T_int))` here to prevent this compiling if the types aren't equivalent. (Unfortunately he typed `static_cast` when he meant `static_assert`).

Reply:
OK, I will add `static_assert`